endurance-page-cache.php

```diff
@@ -585,19 +595,50 @@ public function purge_cdn_single( $pattern = '' ) {
 		/**
 		 * Ensure that a URI isn't purged more than once per minute.
 		 *
-		 * @param string $value URI being purged
+		 * @param string $uri URI being purged
 		 *
-		 * @return bool
+		 * @return bool True if additional purges should be avoided, false otherwise.
 		 */
-		public function purge_throttle( $value ) {
-			$purged = get_transient( 'epc_purged_' . md5( $value ) );
-			if ( ( true === $purged || in_array( md5( $value ), $this->purged, true ) ) && false === $this->force_purge ) {
-				return true;
+		public function should_throttle( $uri ) {
+
+			$should_throttle = false;
+
+			$this->should_update_purges = true;
+
+			$hash = md5( $uri );
+
+			if ( array_key_exists( $hash, $this->purged ) ) {
+				if ( $this->purged[ $hash ] < time() ) {
+					unset( $this->purged[ $hash ] );
+				} else {
+					$should_throttle = true;
+				}
 			}
-			set_transient( 'epc_purged_' . md5( $value ), time(), 60 );
-			$this->purged[] = md5( $value );
 
-			return false;
+			$this->purged[ $hash ] = time() + 60;
+
+			return $should_throttle;
+		}
+
+		/**
+		 * Actions that should take place when the page is done loading.
+		 */
+		public function shutdown() {
+			if ( $this->should_update_purges ) {
+				$purged = array_filter( $this->purged, array( $this, 'filter_expired_timestamps' ) );
+				set_transient( 'epc_purged', $purged, 60 );
+			}
+		}
+
+		/**
+		 * Filters an array of timestamp values and removes items that have expired.
```

Comment:
Is this code comment meant for another function? It's how this helper is used, but not what this method is actually doing -- testing timestamp against present.

Reply:
Agreed. Should probably move this comment up above to line 628 and then clarify what this function specifically does.